x/staking/keeper/slash.go

```diff
@@ -373,11 +373,8 @@ func (k Keeper) SlashRedelegation(ctx context.Context, srcValidator types.Valida
 		if err != nil {
 			return math.ZeroInt(), err
 		}
-		sharesToUnbond, err := dstVal.SharesFromTokensTruncated(slashAmount)
-		if err != nil {
-			return math.ZeroInt(), err
-		}
 
+		sharesToUnbond, _ := dstVal.SharesFromTokensTruncated(slashAmount) // ignore the error as it only errors when tokens == 0
```

Comment:
Seems like it would be safer to do the error check but ignore when sharesToUnbond.IsZero(). If that isn't the case, it could be some other error. This is more future proof

Reply:
Changed it to 

```go
		sharesToUnbond, err := dstVal.SharesFromTokensTruncated(slashAmount) // ignore the error as it only errors when tokens == 0
		if sharesToUnbond.IsZero() {
			continue
		} else if err != nil {
			return math.ZeroInt(), err
		}
```

although in the future we should stop returning an error in `SharesFromTokensTruncated`, because we are only handling the error if we get a value != 0 and an error.